test/test_modelzoo.py

```diff
@@ -0,0 +1,411 @@
+        file.write(line + '\n')
+
+  def generate_report(self):
+    """Generates the report file."""
+    if _CFG['verbose']:
+      print('Writing {}{}\n'.format(_CFG['report_filename'],
+                                    ' (dry_run)' if _CFG['dry_run'] else ''))
+    self._report('*Report generated at {}{}.*'.format(
+        datetime.datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ'),
+        _CFG['github_actions_md']))
+
+    self._report('\n## Environment')
+    self._report('Package | Version')
+    self._report('---- | -----')
+    self._report('Platform | {}'.format(platform.platform()))
+    self._report('Python | {}'.format(sys.version.replace('\n', ' ')))
+    self._report('onnx | {}'.format(onnx.__version__))
+    self._report('onnx-tf | {}'.format(_CFG['onnx_tf_version_md']))
+    self._report('tensorflow | {}'.format(tf.__version__))
+
+    self._report('\n## Summary')
+    self._report('Value | Count')
+    self._report('---- | -----')
+    self._report('Models | {}'.format(self.model_count))
+    self._report('Total | {}'.format(self.total_count))
+    self._report(':heavy_check_mark: Passed | {}'.format(self.pass_count))
+    self._report(':warning: Limitation | {}'.format(self.warn_count))
+    self._report(':x: Failed | {}'.format(self.fail_count))
+    self._report(':heavy_minus_sign: Skipped | {}'.format(self.skip_count))
+
+    self._report('\n## Details')
+    self._report('\n'.join(self.details))
+    self._report('')
+
+  def summary(self):
+    """Returns report summary (counts, report location) as a string."""
+    return ('Total: {}, Passed: {}, Limitation: {}, Failed: {}, '
+            'Skipped: {}\nReport: {}{}').format(
+                self.total_count, self.pass_count, self.warn_count,
+                self.fail_count, self.skip_count, _CFG['report_filename'],
+                ' (dry_run)' if _CFG['dry_run'] else '')
+
+
+def _pull_model_file(file_path):
+  # - If the model file is around ~130B, it's just a file pointer.
+  #   We'll download the file to test, then delete it afterwards
+  #   to minimize disk utilization (important in CI environment).
+  # - If you previously downloaded the file, the file will remain
+  #   in place after processing. In your local environment, make
+  #   sure to pull the models you test often to avoid repetitive
+  #   downloads.
+  model_path = os.path.join(_CFG['models_dir'], file_path)
+  file_size = os.stat(model_path).st_size
+  pulled = False
+  if file_size <= 150:
+    # need to pull the model file on-demand using git lfs
+    if _CFG['verbose']:
+      print('Pulling {}{}'.format(file_path,
+                                  ' (dry_run)' if _CFG['dry_run'] else ''))
+    if not _CFG['dry_run']:
+      cmd_args = 'git lfs pull -I {} -X ""'.format(file_path)
+      subprocess.run(cmd_args,
+                     cwd=_CFG['models_dir'],
+                     shell=True,
+                     check=True,
+                     stdout=subprocess.DEVNULL)
```

Comment:
Do you expect a developer to run test_modelzoo.py in their own environment? If so, we should consider adding instructions on 1. getting git lfs installed, which is required for this python subprocess to work, 2. getting model zoo repo cloned into the current direct, which is also required when going with the default 'models' sub folder.